dev/release/src/campaigns.ts

```diff
@@ -0,0 +1,127 @@
+import { CreatedChangeset } from './github'
+import { readLine } from './util'
+import YAML from 'yaml'
+import execa from 'execa'
+import fetch from 'node-fetch'
+
+// https://about.sourcegraph.com/handbook/engineering/deployments/instances#k8s-sgdev-org
+const DEFAULT_SRC_ENDPOINT = 'https://k8s.sgdev.org'
+
+export interface SourcegraphAuth {
+    SRC_ENDPOINT: string
+    SRC_ACCESS_TOKEN: string
+    [index: string]: string
+}
+
+export async function sourcegraphAuth(): Promise<SourcegraphAuth> {
+    return {
+        SRC_ENDPOINT: DEFAULT_SRC_ENDPOINT,
+        SRC_ACCESS_TOKEN: await readLine('k8s.sgdev.org src-cli token: ', '.secrets/src-cli.txt'),
+    }
+}
+
+export interface CampaignOptions {
+    name: string
+    description: string
+    namespace: string
+    auth: SourcegraphAuth
+}
+
+export function releaseTrackingCampaign(version: string, auth: SourcegraphAuth): CampaignOptions {
+    return {
+        name: `release-sourcegraph-${version}`,
+        description: `Track publishing of sourcegraph@${version}`,
+        namespace: 'sourcegraph',
+        auth,
+    }
+}
+
+interface CampaignSpec {
+    name: string
+    description: string
+    importChangesets: { repository: string; externalIDs: number[] }[]
+}
+
+async function applyCampaign(campaign: CampaignSpec, options: CampaignOptions): Promise<string> {
+    // our version of YAML (<v2) does not seem to support a replacer, so we just do it in JSON,
+    // and the backend does not accept null values - https://github.com/sourcegraph/sourcegraph/issues/15045
+    const campaignYAML = YAML.stringify(
+        JSON.parse(JSON.stringify(campaign, (key, value) => (value === null ? undefined : value))) // eslint-disable-line @typescript-eslint/no-unsafe-return
+    )
+    console.log(`Rendered campaign spec:\n\n${campaignYAML}`)
+    const campaignScript = `set -ex
+
+(
+cat <<EOF
+${campaignYAML}
+EOF
+) | src campaign apply \\
+    -namespace ${options.namespace} \\
+    -f -`
+    await execa('bash', ['-c', campaignScript], { stdio: 'inherit', env: options.auth })
```

Comment:
Feel free to ignore me, because I don't know `execa`, but can't you do this:

```suggestion
    await execa('src', ['campaign', 'apply', '-namespace', options.namespace], {
        stdio: 'inherit',
        input: campaignYAML,
        env: options.auth,
    })
```

Reply:
Well seems you know it better than me 😅 I had to fiddle with this a bit since `stdio: inherit` would not work with this for some reason, but quite a bit cleaner this way! 5a1adb91b6a1f07559d0bc271b8f7146f7c89822